packages/services/api/src/modules/schema/providers/schema-version-helper.ts

```diff
@@ -332,8 +333,29 @@ export class SchemaVersionHelper {
     return schemaVersion.isComposable && schemaVersion.hasContractCompositionErrors === false;
   }
 
+  /**
+   * There's a possibility that the composite schema contains type extensions.
+   * This is a problem, because other parts of the system may expect it to be clean from type extensions.
+   *
+   * This function will also check for type system extensions and merge them into matching definitions.
+   */
+  private autoFixCompositeSchemaAst(schemaAst: DocumentNode): DocumentNode {
```

Comment:
I think it's a little confusing having `autoFixCompositeSchemaSdl` and `autoFixCompositeSchemaAst` but they do very different things.

Reply:
I only followed the convention there. The original autoFix* is specific to federation, but in future it could do many other things.

Reply:
@jdolle I removed the new method and apply its logic to the original one. This way whenever we access composite schema, we will look for type extensions.